drltrace/scripts/headers_parser.py

```diff
@@ -0,0 +1,176 @@
+    @out - a list of WinAPI function prototypes
+    '''
+
+    list_of_results = list()
+    for header_content in headers_content:
+        for idx, line in enumerate(header_content):
+            if entry_name in line:
+                ''' Parse line, look for separate names '''
+                if "(" in line and not "#define" in line and not "return " in line\
+                   and not "//" in line and not "/*" in line and not "*/" in line\
+                   and not "=" in line and not "STDMETHOD" in line\
+                   and not "EXTERN_GUID" in line:
+
+                    final_line = tuple()
+                    tmp_idx = idx
+                    #take return type, it can sit before function or at the same line
+                    while tmp_idx > 1:
+                        tmp_idx = tmp_idx - 1 # looking back
+                        if header_content[tmp_idx] == '\n' or ';' in header_content[tmp_idx]\
+                           or "#" in header_content[tmp_idx]:
+                            break
+                        if header_content[tmp_idx].startswith("\\"): # ignore comments
+                            continue
+                        final_line = final_line + (header_content[tmp_idx],)
+
+                    tmp_idx = idx
+                    # take arguments if exist
+                    while tmp_idx < len(header_content):
+                        if not header_content[tmp_idx].startswith("//"):
+                            comment_idx = header_content[tmp_idx].find("//")
+                            new_line = header_content[tmp_idx]
+                            if comment_idx: # remove comment after an argument
+                                new_line = header_content[tmp_idx][:comment_idx]
+                            final_line = final_line + (new_line,)
+                        if ";" in header_content[tmp_idx]:
+                            break
+                        tmp_idx = tmp_idx + 1
+                    # we have a limitation for the prototype length (e.g. no more than
+                    # 70 elements).
+                    if len(final_line) > 70  or "{" in final_line:
+                        continue
+                    found = 0
+                    '''for element in final_line:
+                        element = element.replace("(", "")
+                        if entry_name == element:
+                            found = 1
+                    if found != 1:
+                        continue'''
+                    list_of_results.append(final_line)
+    results = ""
+    duplicates = 0
+    for entry in list_of_results:
+        results += merge_entries(entry, duplicates, entry_name) + "\n"
+        duplicates = 1
+    return results
+
+if __name__ == "__main__":
+    export_files = [f for f in listdir(EXPORT_PATH) if isfile(join(EXPORT_PATH, f))]
+    headers_content = parse_headers() # parse all headers
+
+    for idx, file in enumerate(export_files):
```

Comment:
Good question, I'm producing a list of exported functions using a simple IDAPython script. The script requires IDA Disassembler, IDAPython and several other third-party libraries to be installed. Do we want this script to be checked out ? I'm not sure about licenses of these third-party libraries.